### PR TITLE
initial testdata script

### PR DESCRIPTION
**Pull request by flatberg**

### CODE REVIEW

File: testdata/create_testdata.py [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+            self._fastq_dir = os.path.join(self.dirname, 'raw_fastq_{}'.format(self._gcf_number))
+        else:
+            self._fastq_dir = os.path.join(self.dirname, self._gcf_number)
+        if not os.path.exists(self._fastq_dir):
+            raise ValueError('Missing fastq dir. Expected {}'.format(self._fastq_dir)) 
+
+        df = pd.read_csv(os.path.join(self.dirname, '{}_samplesheet.tsv'.format(self._gcf_number)), sep='\t')
+        self.fastq_files = {}
+        for sample in df.Sample_ID:
+            self.fastq_files[sample] = glob.glob1(self._fastq_dir, '{}*.fastq.gz'.format(sample))
```
Comment (gsmashd) [inline testdata/create_testdata.py:113]: Denne failer på single cell. Eks GCF-2019-668:

>>> import glob
>>> glob.glob1("/mnt/bfq/output/191112_NB501038_0178_AHHWGKBGXC/GCF-2019-668","APP_LEC_LII_P45_rep1*.fastq.gz")
[]

Kan endres til 
if singlecell:
    self.fastq_files[sample] = glob.glob1(os.path.join(self._fastq_dir, sample), '{}*.fastq.gz'.format(sample))

File: testdata/create_testdata.py [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+            fq_files = self.fastq_files[sample]
+            for fq_basename in fq_files:
+                src = os.path.join(self._fastq_dir, fq_basename)
+                if self.pipeline == 'single-cell':
+                    new_baseneame = fq_basename
+                else:
+                    read_num = 'R1' if fq_basename.endswith('_R1.fastq') else 'R2'
+                    new_basename = '{}_L001_{}_001.fastq'.format(sample, read_num)
+                    dst = os.path.join(fastq_dir_output, new_basename)
+                    cmd = 'zcat {} | seqkit sample -n {} -s 123456 > {}'.format(src, n_reads, dst)
```
Comment (gsmashd) [inline testdata/create_testdata.py:175]: Egentlig ment å bruke subset_fastq definert i linje 200 her? Finner ingen referanser til subset_fastq ellers.

File: testdata/create_testdata.py (hunk 1 above)
Comment (flatberg) [reply testdata/create_testdata.py:113]: fixed

File: testdata/create_testdata.py [hunk 3]
```diff
@@ -0,0 +1,223 @@
...
+    
+    """
+    output = []
+    with open(input_fn) as fh:
+        for i, line in enumerate(fh):
+            if line.startswith('[Data]'):
+                header = fh.readline().strip('\n').split(',')
+                sample_id_index = header.index('Sample_ID')
+                header_line = i +1
+    with open(input_fn) as fh:
```
[the remark this answers is not in the document]
Comment (flatberg) [reply testdata/create_testdata.py:47]: skipped

File: testdata/create_testdata.py (hunk 2 above)
Comment (flatberg) [reply testdata/create_testdata.py:175]: nah

Comment (flatberg) [reply testdata/create_testdata.py:175]: removed subset_fastq function

File: testdata/create_testdata.py [hunk 4]
```diff
@@ -110,8 +110,8 @@ def _inspect(self):
         df = pd.read_csv(os.path.join(self.dirname, '{}_samplesheet.tsv'.format(self._gcf_number)), sep='\t')
         self.fastq_files = {}
         for sample in df.Sample_ID:
-            self.fastq_files[sample] = glob.glob1(self._fastq_dir, '{}*.fastq.gz'.format(sample))
-                
+            self.fastq_files[sample] = glob.glob1(os.path.join(self._fastq_dir, sample), '{}*.fastq.gz'.format(sample))
```
Comment (gsmashd) [inline testdata/create_testdata.py:113]: Denne vil kun fungere for single cell. Hva med:
```python
search_path = os.path.join(self._fastq_dir, sample) if PIPELINE_MAP.get(self.libprep, None) == "single-cell" else self._fastq_dir
self.fastq_files[sample] = glob.glob1(search_path, '{}*.fastq.gz'.format(sample))
```

**Review by gsmashd — CHANGES_REQUESTED** [review]

**Review by gsmashd — APPROVED** [review]

**Pull request merged**